src/mscorlib/shared/System/IO/PathInternal.cs

```diff
@@ -10,8 +10,7 @@ internal static partial class PathInternal
         /// <summary>
         /// Returns true if the path ends in a directory separator.
         /// </summary>
-        internal static bool EndsInDirectorySeparator(string path) =>
-            !string.IsNullOrEmpty(path) && IsDirectorySeparator(path[path.Length - 1]);
+        internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path) => IsDirectorySeparator(path[path.Length - 1]);
```

Comment:
Does this need to handle `path.Length == 0` ?

Reply:
Not currently, but I think it is better to have it.